src/vs/platform/instantiation/test/common/instantiationServiceMock.ts

```diff
@@ -129,6 +129,10 @@ export class TestInstantiationService extends InstantiationService {
 	override createChild(services: ServiceCollection): TestInstantiationService {
 		return new TestInstantiationService(services, false, this);
 	}
+
+	dispose() {
+		sinon.restore();
```

Comment:
I think calling restore on sinon here might not be correct as this will restore all sinon stubs that this service does not own.